tt_eager/tt_dnn/op_library/sliding_window_op_infra/tt_py_untilize_with_halo.py

```diff
@@ -213,86 +164,65 @@ def set_op_configs(
                 )
                 shard_grid = ttl.tensor.CoreRangeSet({core_range})
 
-            def gen_config_tt_tensors_uint16(config_list_uint16: list, toprint=False):
-                config_size = len(config_list_uint16)
-                if config_size == 0:
-                    # return dummy tensor
-                    output = ttl.tensor.Tensor(
-                        [0, 0], [1, 1, 1, 2], ttl.tensor.DataType.BFLOAT16, ttl.tensor.Layout.ROW_MAJOR
-                    )
-                    if device is not None:
-                        output = output.to(device)
-                    return output
-                assert config_size % 2 == 0  ## each config is a tuple of (start, size)
-                assert config_size % num_cores_nhw == 0
+            def gen_per_core_gather_data_uint16_tensor(config: list):
+                assert type(config) is list
+                if block_sharding:
+                    assert len(config) == num_cores_w, f"{len(config)} {num_cores_w}"
```

Comment:
Yeah for block_sharded `len(config) == num_cores_w == num_cores_nhw`, `num_cores_w` feels a bit more explicit since this is only true if block sharding.